providers/implementations/signature/ecdsa_sig.c

```diff
@@ -66,8 +73,18 @@ typedef struct {
     OSSL_LIB_CTX *libctx;
     char *propq;
     EC_KEY *ec;
-    char mdname[OSSL_MAX_NAME_SIZE];
+    int operation;
 
+    /*
+     * Flag to determine if a full sigalg is run (1) or if a composable
+     * signature algorithm is run (0).
+     *
+     * When a full sigalg is run (1), this currently affects the following
+     * other flags, which are to remain untouched after their initialization:
+     *
+     * - flag_allow_md (initialized to 0)
+     */
+    unsigned int flag_sigalg : 1;
```

Comment:
"sigalg" is a term I borrowed from libssl code, see for example `SSL_get_shared_sigalgs()`.  Not sure that "full" is meaningful enough, but I could go with "composite", or if we shall lean on standards, "sigscheme" (see PKCS#1).

But yeah, I generally agree, we have some pretty...  "interesting" nomenclature throughout the libraries.

Reply:
signature algorithm is not far from just algorithm, so it cant be used to determine the difference. (This did my head in reading the code because I didnt write it :))

Reply:
Not sure what conclusion we came to...